app/src/main/java/org/p2p/wallet/auth/ui/animationscreen/AnimationProgressFragment.kt

```diff
@@ -0,0 +1,122 @@
+package org.p2p.wallet.auth.ui.animationscreen
+
+import android.os.Bundle
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.core.view.isVisible
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.lifecycleScope
+import androidx.transition.TransitionManager
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.p2p.wallet.R
+import org.p2p.wallet.common.mvp.BaseFragment
+import org.p2p.wallet.databinding.FragmentAnimationProgressBinding
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+import kotlin.time.Duration.Companion.seconds
+
+private const val ARG_TITLE_RES = "ARG_TITLE_RES"
+
+class AnimationProgressFragment : BaseFragment(R.layout.fragment_animation_progress) {
+
+    companion object {
+        private val fragmentTag = AnimationProgressFragment::class.java.name
+        private var isAnimating = false
+
+        fun create(loadingTitleRes: Int) = AnimationProgressFragment().withArgs(ARG_TITLE_RES to loadingTitleRes)
+
+        fun show(fragmentManager: FragmentManager, isCreation: Boolean) {
+            if (!isAnimating) {
+                isAnimating = true
+                val loadingTitleRes = if (isCreation) {
+                    R.string.onboarding_loading_creating_title
+                } else {
+                    R.string.onboarding_loading_recovery_title
+                }
+                fragmentManager.beginTransaction()
+                    .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
+                    .add(
+                        android.R.id.content,
+                        create(loadingTitleRes),
+                        fragmentTag
+                    )
+                    .commit()
+            }
+        }
```

Comment:
I am pretty sure we can do better. 
For example launch the fragment in the root activity and hide there as well. 
We can remove static boolean fields from here as well.


Anyway this looks not good

Reply:
sorry but not =(
because when we call this method check in fragment manager that there is already Fragment exists not working as Fragment manager creating it toooooooo long
so if we call show some times faster then it is 
we will have 2-3 fragments crated and added =(

i spend some time on fixing this 

if you or @gslevinkov have some clear and good solution i will implement this =( but FragmentManager does not have good check to find fragment same time it is added to FragmentTransaction.... =(